packages/ui/src/Checkbox.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { AriaAttributes, ChangeEvent, FC, FocusEvent, useRef } from 'react'
+import styles from './Checkbox.module.scss'
+import { Check, Minus } from 'react-feather'
+import { DataTestProp } from '@hazelcast/helpers'
+import classNames from 'classnames'
+import { Error, errorId } from './Error'
+import { Help, helpTooltipId } from './Help'
+import { v4 as uuid } from 'uuid'
+
+type CheckboxCoreProps = {
+  name: string
+  value?: string
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void
+  error?: string
+  checked?: boolean
```

Comment:
When we pass `undefined` we make the checkbox uncontrolled
```suggestion
  checked: boolean
```